examples/xpress_tests/xpress_use.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright Artelys for RTE.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This example that shows how to use Xpress Solver.
+
+#include <cstdlib>
+
+#include "ortools/base/commandlineflags.h"
+#include "ortools/base/init_google.h"
+#include "ortools/base/logging.h"
+#include "ortools/linear_solver/linear_solver.h"
+
+using namespace operations_research;
+
+/**
+ * This method shows two ways to initialize a Xpress solver instance.
+ * Two environment variables are used to specify the Xpress installation paths:
+ *  * XPRESSDIR : Path to the Xpress root directory (containing bin and lib folders)
+ *  * XPRESS : Path to the directory containing Xpress license
+ */
+void useXpressSolver(bool solveAsMip, bool useFactory) {
+  std::unique_ptr<MPSolver> solver = nullptr;
+  if (useFactory) {
+    /* This is the preferred way as the program won't stop if anything went
+       wrong. In such a case, `solver` will take value `nullptr` */
+    std::string xpressName = (solveAsMip ? "XPRESS" : "XPRESS_LP");
+    solver.reset(MPSolver::CreateSolver(xpressName));
```

Comment:
Make sure you do not crash if you cannot load the dll/license.

Reply:
Actually, remove and add the xpress_flags in the official examples (integer_programming.cc, .py, linear_programming, java, .NET)

Reply:
This is an important point:
is it ok for you to have some tests, (disabled by default), that can pass only if the xpress_library/licence are found ?
This is the strategy that I proposed in this PR. I introduced a CMake variable `BUILD_XPRESS_TEST_AND_EXAMPLES` (OFF by default) that can activate some tests that pass only if Xpress (dll/licence) is found.
If this strategy is not ok for you, I can change it, remove those tests from this PR  (or change them in order to never crash) and only have some generic tests.